pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -369,6 +369,20 @@ def configure_ddp(self) -> None:
         )
         self._register_ddp_hooks()
 
+    @property
+    def should_rank_save_checkpoint(self) -> bool:
+        """Returns whether the checkpoint should be saved (rank based)"""
+        if self.num_nodes > 1:
+            sync_dir = None
+            if self.is_global_zero:
+                sync_dir = tempfile.mkdtemp()
+            self.barrier()
+            sync_dir = self.broadcast(sync_dir)
+            shared_file_system = self.all_gather(torch.tensor(int(os.path.exists(sync_dir)))).sum() == self.world_size
```

Comment:
the os.path.exists check will not work or a remote filesystem (s3, gcs) used with fsspec